### ISSUE TITLE
A question about the usage.

### ISSUE THREAD

**Issue by EmilyWang90** [mk-minchul/AdaFace#5]
Hello,

Thank you for presenting the solid and wonderful work. I have a small question about the usage.

cosine_with_margin = adaface(embbedings, norms, labels)
loss = torch.nn.CrossEntropyLoss()(cosine_with_margin, labels)

After I read the code in head.py. I thought the return value of cosine_with_margin is the variant of the logits. In other words, it is the output of the adaptive margin function in the paper. Has it been passed into the softmax function? Why we can input it into the CrossEntropyLoss and compare it with ground truth directly?

 I'm looking forward to your reply. Thank you very much.

**Comment by mk-minchul** [comment]
Hi EmilyWang90
The output cosine_with_margin is equivalent to the f() is equation 18 of the main paper. So if you place f() in equation 1 of the main paper, you can see that it is the input to the cross entropy loss (before having applied the softmax fn).
